src/DataStructures/Variables.hpp

```diff
@@ -292,6 +292,40 @@ class Variables<tmpl::list<Tags...>> {
   }
   /// @}
 
+  /// @{
+  /// \brief Assign a subset of the `Tensor`s from another Variables or a
+  /// tuples::TaggedTuple. Any tags that aren't in both containers are
+  /// ignored.
+  ///
+  /// \note There is no need for an rvalue overload because we need to copy into
+  /// the contiguous array anyway
+  template <typename... SubsetOfTags>
+  void assign_overlapping_subset(
```

Comment:
This may be a naive question, but it seems like this new function is just a slightly more general `assign_subset` (it doesn't require all of `SubsetOfTags...` to be in `Tags...`). Why not just get rid of the restriction in `assign_subset` and change the functionality there?

Reply:
I think the original reason for having the restriction was that you'd get errors if some of the tags didn't match rather than them silently being ignored. I don't really feel strongly either way.

Reply:
I think it'd be ok to modify `assign_subset` to remove the restriction of having the tags match, but I also understand it might be nice to keep the compile time check. Up to you. If you decide to leave it as it, add some docs to `assign_subset` saying that all tags in `SubsetOfTags` must be in `Tags`.

Reply:
Sounds good!

@wthrowe @kidder would you prefer a new function or just extend the existing?

Reply:
I don't have a preference